app/controllers/image_uploads_controller.rb

```diff
@@ -30,4 +33,20 @@ def create
       format.json { render json: { link: link }, status: 200 }
     end
   end
+
+  def limit_uploads
+    count = Rails.cache.read("#{current_user.id}_image_upload")
+
+    if count.nil?
+      count = 1
+    else
+      count += 1
+    end
+
+    if count == 10
+      Rails.cache.write("#{current_user.id}_image_upload", count, expires_in: 30.seconds)
+    else
+      Rails.cache.write("#{current_user.id}_image_upload", count)
```

Comment:
Should the `cache.write` in the `else` statement have an explicit expiration time?

Reply:
Oh yeah, it makes sense to have everything start over after 30 seconds. That would actually clean this up too, thanks!

Reply:
And thanks for the tip that nil converts to 0!